extensions/amp-viqeo/0.1/amp-viqeo.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-unused-vars */
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {Layout, isLayoutSizeDefined} from '../../../src/layout';
+import {setStyle} from '../../../src/style';
+import {user} from '../../../src/log';
+
+class AmpViqeo extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+  }
+
+  /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://static.viqeo.tv', opt_onLayout);
+    this.preconnect.url('https://stage.embed.viqeo.tv', opt_onLayout);
+  }
+
+  /**
+   * @param {!Layout} layout
+   * @return {boolean}
+   * @override
+   */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    return null;
+  }
+
+
+  /** @override */
+  layoutCallback() {
+    const videoId = user().assert(
+        this.element.getAttribute('data-videoid'),
+        'The data-videoid attribute is required for <amp-viqeo> %s',
+        this.element);
+
+    const profileId = user().assert(
+        this.element.getAttribute('data-profileid'),
+        'The data-profileid attribute is required for <amp-viqeo> %s',
+        this.element);
+
+    const kindIsProd = this.element.getAttribute('data-kind') !== 'stage';
+
+    const iframeStyle = this.element.getAttribute('data-iframe-style')
+      || 'position: absolute';
+    const iframeHeight = this.element.getAttribute('data-iframe-height')
```

Comment:
Layout is AMP is fully controlled by out layout system, authors should not be able to ( and is pretty pointless anyway ) to change the style of the iframe.